tests/vmi_networking_test.go

```diff
@@ -647,22 +646,18 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			err = tests.CheckForTextExpecter(dhcpVMI, []expect.Batcher{
 				&expect.BSnd{S: "\n"},
 				&expect.BExp{R: "\\#"},
-				&expect.BSnd{S: "sudo dhclient -1 -r -d eth0\n"},
+				&expect.BSnd{S: "dhclient -1 -r -d eth0\n"},
 				&expect.BExp{R: "\\#"},
-				&expect.BSnd{S: "sudo dhclient -1 -sf /usr/bin/env --request-options subnet-mask,broadcast-address,time-offset,routers,domain-search,domain-name,domain-name-servers,host-name,nis-domain,nis-servers,ntp-servers,interface-mtu,tftp-server-name,bootfile-name eth0 | tee /dhcp-env\n"},
+				&expect.BSnd{S: "dhclient -1 -sf /usr/bin/env --request-options subnet-mask,broadcast-address,time-offset,routers,domain-search,domain-name,domain-name-servers,host-name,nis-domain,nis-servers,ntp-servers,interface-mtu,tftp-server-name,bootfile-name eth0 | tee /dhcp-env\n"},
 				&expect.BExp{R: "\\#"},
 				&expect.BSnd{S: "cat /dhcp-env\n"},
```

Comment:
Lets fix it with less noise:
```
grep -q "new_tftp_server_name=tftp.kubevirt.io" /dhcp-env
```
Then check success/failure.

What say you?

Reply:
yes, that's nicer, I will try it.

Reply:
That's how it looks with the purposed fix. Very long! I'm not sure it is better. What do you think?
'''
	err = tests.CheckForTextExpecter(dhcpVMI, []expect.Batcher{
				&expect.BSnd{S: "\n"},
				&expect.BExp{R: "\\#"},
				&expect.BSnd{S: "dhclient -1 -r -d eth0\n"},
				&expect.BExp{R: "\\#"},
				&expect.BSnd{S: "dhclient -1 -sf /usr/bin/env --request-options subnet-mask,broadcast-address,time-offset,routers,domain-search,domain-name,domain-name-servers,host-name,nis-domain,nis-servers,ntp-servers,interface-mtu,tftp-server-name,bootfile-name eth0 | tee /dhcp-env\n"},
				&expect.BExp{R: "\\#"},
				&expect.BSnd{S: "grep -q 'new_tftp_server_name=tftp.kubevirt.io' /dhcp-env\n"},
				&expect.BExp{R: "\\#"},
				&expect.BSnd{S: "echo $?\n"},
				&expect.BExp{R: tests.RetValue("0", "\\$ ")},
				&expect.BSnd{S: "grep -q 'new_bootfile_name=config' /dhcp-env\n"},
				&expect.BExp{R: "\\#"},
				&expect.BSnd{S: "echo $?\n"},
				&expect.BExp{R: tests.RetValue("0", "\\$ ")},
				&expect.BSnd{S: "grep -q 'new_ntp_servers=127.0.0.1 127.0.0.2' /dhcp-env\n"},
				&expect.BExp{R: "\\#"},
				&expect.BSnd{S: "echo $?\n"},
				&expect.BExp{R: tests.RetValue("0", "\\$ ")},
				&expect.BSnd{S: "grep -q 'new_unknown_240=private.options.kubevirt.io' /dhcp-env\n"},
				&expect.BExp{R: "\\#"},
				&expect.BSnd{S: "echo $?\n"},
				&expect.BExp{R: tests.RetValue("0", "\\$ ")},
			}, 15)
'''

Reply:
If this is about saving lines, then you could use an regex OR to grep them all in one shot.
BTW, you could use a similar expect expression if you lose the `-q`, that may shorten it as well.

Leaving it up to you.

Reply:
I will try - "grep -q 'new_unknown_240=private.options.kubevirt.io' /dhcp-env; echo $?\n"

Reply:
done

Reply:
Maybe I'm missing something but why are we not using `ExpectBatchWithValidatedSend()` here?

Reply:
> done

I don't see it, is github messing with me?

Reply:
`CheckForTextExpecter` is using.
We use "Done" to mention it was implemented, and that you will see it in the next push.
Done & Push are raceful :)

Reply:
`tests.CheckForTextExpecter` is using `ExpectBatchWithValidatedSend`

Reply:
Ah, then why all the prompt noise? Doesn't `ExpectBatchWithValidatedSend` already takes care of expecting them?

Reply:
>> We use "Done" to mention it was implemented, and that you will see it in the next push.
Done & Push are raceful :) 
Yes, exactly:)

Reply:
> Ah, then why all the prompt noise? Doesn't `ExpectBatchWithValidatedSend` already takes care of expecting them?

The prompt will be removed once the `"utils.RetValue" no need to pass prompt`
 [PR](https://github.com/kubevirt/kubevirt/pull/3956) is merged. (I don't know which one of them will be merged first, but the second one will have a conflict).

`ExpectBatchWithValidatedSend` doesn't take care of expecting prompt. It verifies the expected result came after the sent command (doing so reduces the need waiting for the prompt but it is better to do it anyway to be on the safe side).